mungegithub/mungers/submit-queue-batch.go

```diff
@@ -0,0 +1,282 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mungers
+
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+	githubapi "github.com/google/go-github/github"
+	"k8s.io/contrib/mungegithub/github"
+)
+
+// xref k8s.io/test-infra/prow/cmd/deck/jobs.go
+type ProwJob struct {
+	Type    string `json:"type"`
+	Repo    string `json:"repo"`
+	Refs    string `json:"refs"`
+	State   string `json:"state"`
+	Context string `json:"context"`
+}
+
+// getSuccessfulBatchJobs reads test results from Prow and returns
+// all batch jobs that succeeded for the current repo.
+func getSuccessfulBatchJobs(repo, url string) ([]ProwJob, error) {
+	resp, err := http.Get(url)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+
+	allJobs := []ProwJob{}
+	err = json.Unmarshal(body, &allJobs)
+	if err != nil {
+		return nil, err
+	}
+
+	jobs := []ProwJob{}
+	for _, job := range allJobs {
+		if job.Repo == repo && job.Type == "batch" && job.State == "success" {
+			jobs = append(jobs, job)
+		}
+	}
+	return jobs, nil
+}
+
+type BatchPull struct {
+	Number int
+	Sha    string
+}
+
+type Batch struct {
+	BaseName string
+	BaseSha  string
+	Pulls    []BatchPull
+}
+
+// batchRefToBatch parses a string into a Batch
+func batchRefToBatch(batchRef string) (Batch, error) {
+	batch := Batch{}
+	for i, ref := range strings.Split(batchRef, ",") {
+		parts := strings.Split(ref, ":")
+		if len(parts) != 2 {
+			return Batch{}, errors.New("bad batchref")
+		}
+		if i == 0 {
+			batch.BaseName = parts[0]
+			batch.BaseSha = parts[1]
+		} else {
+			num, err := strconv.ParseInt(parts[0], 10, 32)
+			if err != nil {
+				return Batch{}, err
+			}
+			batch.Pulls = append(batch.Pulls, BatchPull{int(num), parts[1]})
+		}
+	}
+	return batch, nil
+}
+
+// getCompleteBatches returns a list of Batches that passed all
+// required tests.
+func (sq *SubmitQueue) getCompleteBatches(jobs []ProwJob) []Batch {
+	batchContexts := make(map[string]map[string]interface{})
+	for _, job := range jobs {
+		if _, ok := batchContexts[job.Refs]; !ok {
+			batchContexts[job.Refs] = make(map[string]interface{})
+		}
+		batchContexts[job.Refs][job.Context] = nil
+	}
+	batches := []Batch{}
+	for batchRef, contexts := range batchContexts {
```

Comment:
In this loop:
- match is initialized to true
- it can go to false, but can't go back to true
- if match is false, it doesn't have any side effects

Could you:
- remove `match := true`
- replace `match = false` with `continue`
- remove `if match { }` ?